default/python/AI/SaveGameManager/_decoder.py

```diff
@@ -0,0 +1,182 @@
+"""This module defines the decoding for the FreeOrion AI savegames.
+
+The decoder is subclassed from the standard library json decoder and
+uses its string parsing. However, the resulting objects will be interpreted
+differently according to the encoding used in FreeOrion AI savegames.
+
+The decoder will only load trusted classes as defined in _definitions.py,
+if an unknown/untrusted object is encountered, it will raise a InvalidSaveGameException.
+
+When classes are loaded, their __setstate__ method will be invoked if available or
+the __dict__ content will be set directly. It is the responsiblity of the trusted classes
+to provide a __setstate__ method to verify and possibly sanitize the content of the passed state.
+
+Example usage:
+    Importing this module directly is not advised.
+    Use the exported functions when importing the package instead.
+"""
+import json
+
+import EnumsAI
+from freeorion_tools import profile
+
+from _definitions import *
+
+
+@profile
+def load_savegame_string(string):
```

Comment:
It is good place to add type hint for returning value.  PyCharm will have autocomplete on aistate.

"""
rtype: AIState
"""